dashboards-observability/public/components/application_analytics/components/application.tsx

```diff
@@ -192,6 +196,26 @@ export function Application(props: AppDetailProps) {
     );
   };
 
+  const getPanel = () => {
+    return (
+      <CustomPanelView
+        panelId={''}
+        http={http}
+        pplService={pplService}
+        chrome={chrome}
+        parentBreadcrumb={[parentBreadcrumb]}
+        // App analytics will not be renaming/cloning/deleting panels
+        renameCustomPanel={() => undefined}
+        cloneCustomPanel={():Promise<string> => Promise.reject()}
+        deleteCustomPanel={():Promise<string> => Promise.reject()}
```

Comment:
why clone/delete are returning `Promise.reject()` but rename is returning `undefined`?

Reply:
They seem to have different return types, clone/delete required a Promise to be returned but rename allowed undefined. I can change them all to be `Promise.reject()` if it's confusing. The buttons to call those functions will be hidden from the UI.

Reply:
Are you planning to remove these buttons in panels or put some bool flags in CustomPanelsView component.

Reply:
The second option, I use the page prop to decide whether to show the buttons or not.